rasa/nlu/training_data/formats/luis.py

```diff
@@ -18,13 +19,24 @@ def read_from_json(self, js: Dict[Text, Any], **kwargs: Any) -> "TrainingData":
         training_examples = []
         regex_features = []
 
-        # Simple check to ensure we support this luis data schema version
-        if not js["luis_schema_version"].startswith("2"):
+        # Simple check to ensure we support this luis data schema
+        if not (
+            "regex_features" in js
+            and "utterances" in js
+            and "text" in js["utterances"][0]
+            and "intent" in js["utterances"][0]
+            and "entities" in js["utterances"][0]
+        ):
             raise Exception(
-                "Invalid luis data schema version {}, "
-                "should be 2.x.x. "
-                "Make sure to use the latest luis version "
-                "(e.g. by downloading your data again)."
+                "Invalid LUIS data schema structure." "Your file is not supported."
+            )
+
+        version = int(js["luis_schema_version"][0])
+        if version > 5:
```

Comment:
Why the "magic number" 5 is here?
Microsoft LUIS `JSON` version 5 is released and is not compatible with RASA?

Reply:
I guess the actual version is `3.2.0` [link](https://westus.dev.cognitive.microsoft.com/docs/services/5890b47c39e2bb17b84a55ff/operations/5890b47c39e2bb052c5b9c40) right?

Reply:
On 12/09 we exported Microsoft LUIS JSON from its platform and the version in that date was 5.0.0. But today I exported again and they rolled back to version 4.0.0: `{ "luis_schema_version": "4.0.0", ... }`.
This number at line 35 would be the version that is checked for us and it is compatible with RASA.